### ISSUE TITLE
Implementation of Chloride + Water Electrostatics in Reference platform

### ISSUE THREAD

**Issue by zonca** [paesanilab/mbpol_openmm_plugin#50]
@pushpbajaj I prepared the software to support ions, here the parameters I need:

* mass
* charge
* thole parameters (for water I have charge-charge, charge-dipole and 3 types of dipole-dipole)
* polarizability
* damping factor
* any exclusions? does the ion interact with the virtual site of water molecules?

**Comment by zonca** [comment]
For the test case, the best would be to have 3 water molecules and 1 ion.

@pushpbajaj , could you please run a simulation with only electrostatics activated, using 3 water molecules in these positions (any position is fine for the ion):

https://github.com/paesanilab/mb-pol/blob/master/orig_mbpol_cpp/trimer.xyz

And paste the output energy and forces here?

**Comment by pushpbajaj** [comment]
Following are the parameters for chloride: 

Mass = 35.5 g/mol
Charge = -1 (Please check what units are expected by the electrostatics code, you might have to multiply this with a conversion factor)
The C-C, C-D and D-D parameters (I believe the smearing parameters) should remain the same as in water.
Polarizability = 5.3602 Angstrom^3
Damping factor = 5.3602 (Same as the polarizability)
The excluded pairs should be the same as in water. 

I will create a test case and put it up here.

**Comment by pushpbajaj** [comment]
So, using this configuration: 

10
x
O      -1.516074336e+00  -2.023167650e-01   1.454672917e+00
H      -6.218989773e-01  -6.009430735e-01   1.572437625e+00
H      -2.017613812e+00  -4.190350349e-01   2.239642849e+00
O      -1.763651687e+00  -3.816594649e-01  -1.300353949e+00
H      -1.903851736e+00  -4.935677617e-01  -3.457810126e-01
H      -2.527904158e+00  -7.613550077e-01  -1.733803676e+00
O      -5.588472140e-01   2.006699172e+00  -1.392786582e-01
H      -9.411558180e-01   1.541226676e+00   6.163293071e-01
H      -9.858551734e-01   1.567124294e+00  -8.830970941e-01
Cl      0.000000000e+00   0.000000000e+00   0.000000000e+00

The following is the output:
Eelec = -4.880922637e+00
Eind = -3.623251767e+01
Gradients returned from electrostatics : 
grd[0] = 2.613683596e+01
grd[1] = 5.408275264e+00
grd[2] = -1.629517609e+01
grd[3] = -2.216918164e+01
grd[4] = -2.590793612e+01
grd[5] = 6.296385402e+01
grd[6] = 1.517701844e+00
grd[7] = -3.046093356e+00
grd[8] = 3.837297571e+00
grd[9] = 1.636350854e+01
grd[10] = -4.336106706e+00
grd[11] = 1.972968278e+01
grd[12] = -3.640857280e+01
grd[13] = -8.181611462e+00
grd[14] = -2.800077880e+01
grd[15] = -1.656236514e+00
grd[16] = -1.105120684e+00
grd[17] = 7.059863729e-01
grd[18] = 9.977840973e+00
grd[19] = -3.226637538e+01
grd[20] = 8.605903697e+00
grd[21] = -1.536072074e+01
grd[22] = 3.348477786e+01
grd[23] = 8.224734193e+00
grd[24] = -2.036848206e+00
grd[25] = 1.324057610e+01
grd[26] = -9.839995796e+00
grd[27] = 2.363567258e+01
grd[28] = 2.270961449e+01
grd[29] = -4.993150796e+01

Where index 27, 28, 29 correspond to the x, y, z of chloride.

**Comment by zonca** [comment]
@pushpbajaj thanks, here your forces reformatted:

```
2.613683596e+01 5.408275264e+00 -1.629517609e+01
-2.216918164e+01 -2.590793612e+01 6.296385402e+01
1.517701844e+00 -3.046093356e+00 3.837297571e+00
1.636350854e+01 -4.336106706e+00 1.972968278e+01
 -3.640857280e+01 -8.181611462e+00 -2.800077880e+01
 -1.656236514e+00 -1.105120684e+00 7.059863729e-01
 9.977840973e+00 -3.226637538e+01 8.605903697e+00
 -1.536072074e+01 3.348477786e+01 8.224734193e+00
 -2.036848206e+00 1.324057610e+01 -9.839995796e+00
 2.363567258e+01 2.270961449e+01 -4.993150796e+01
```

**Comment by zonca** [comment]
Implemented in #55
